plugins/scale/scale-title-overlay.cpp

```diff
@@ -0,0 +1,326 @@
+#include "scale-title-overlay.hpp"
+
+#include <wayfire/opengl.hpp>
+#include <wayfire/plugins/common/cairo-util.hpp>
+#include <wayfire/plugins/common/simple-texture.hpp>
+
+class view_title_overlay : public wf::custom_data_t
+{
+  protected:
+    wf::cairo_text_t overlay;
+    wf::cairo_text_t::params par;
+    wayfire_view view;
+    /* the transformer we are attached to */
+    wf::scale_transformer& tr;
+    /* if text potentially overflows the available space */
+    bool overflow = false;
+
+    /**
+     * Get the transformed bounding box of the view, including the
+     * current transform
+     */
+    wlr_box get_transformed_bounding_box()
+    {
+        wlr_box box = view->get_wm_geometry();
+        box = tr.trasform_box_without_padding(box);
+        return box;
+    }
+
+    void render_overlay()
+    {
+        wlr_box box = get_transformed_bounding_box();
+        par.max_size = {box.width, box.height};
+        auto res = overlay.render_text(view->get_title().c_str(), par);
+        overflow = res.width > overlay.tex.width;
+        view->damage();
+    }
+
+  public:
+    enum class position
+    {
+        TOP,
+        CENTER,
+        BOTTOM,
+    };
+
+    position pos = position::CENTER;
+
+    GLuint get_texture(float output_scale, wf::geometry_t& geometry)
+    {
+        wlr_box box = get_transformed_bounding_box();
+        int w = overlay.tex.width;
```

Comment:
you could assign these after checking for regeneration of the overlay, then you won't have to re-assign the values.